src/store/recordAction.ts

```diff
@@ -0,0 +1,13 @@
+import * as FullStory from '@fullstory/browser';
+import { Middleware } from 'redux';
+
+const recordAction: Middleware = () => next => action => {
+  if (typeof action !== 'function') {
+    FullStory.log(action.type);
+  } else if (typeof action === 'function') {
+    FullStory.log('Action is a thunk');
```

Comment:
Since thunks should call a regular actions at a later point in time, I'm not sure I understand the value of logging information from a thunk?

Reply:
Some of the research we've done for experience debugging points to two cases that should be interesting are:
1. expected actions not occurring
2. actions occurring out of expected order

I originally did not record thunk actions, but then thought twice about it. Could the absence of an expected thunk action provide meaningful trouble-shooting data to a developer?

Reply:
For the two points, definitely agree that they are the key use cases!

1. expected actions not occurring
Most common async patterns that are driven through thunks tend to follow a 'start' action, followed by a 'failure'/'success' action when the async behavior completes. In this case where an action isn't occurring, it should be able to be seen by finding the 'start' action and then not seeing any follow-up action. 

2. actions occurring out of expected order
I think this is similar to point one, this is definitely possible, but could be found through the timestamps associated with each action.

>  Could the absence of an expected thunk action provide meaningful trouble-shooting data to a developer?
I think the main use case I could see from this is something like the following:

```ts
export const myThunkAction = () => {
  return (dispatch, getState) => {
    // early return causes the actions not to fire
    if (getState().reducer.myProperty) return;
    dispatch(myIntendedActionCreator());
  }
})
```

But, if we were only logging the actions, the absence of the event would also suggest that the early return was happening. The last aspect of this is our inability to associate much more info from this other than the fact that a thunk action was called, because there's no `type` associated with a thunk action.

Reply:
Got it. I'm going to remove the thunk logging as it causes more noise than signal @timchin